lightning/src/chain/chaininterface.rs

```diff
@@ -55,7 +55,7 @@ pub trait ChainWatchInterface: Sync + Send {
 
 	/// Gets the list of transaction indices within a given block that the ChainWatchInterface is
 	/// watching for.
-	fn filter_block(&self, block: &Block) -> Vec<usize>;
+	fn filter_block(&self, header: &BlockHeader, txdata: &[(usize, &Transaction)]) -> Vec<usize>;
```

Comment:
Why define this in terms of pre-filtered data? Seems like you could just pass in a &[Transaction] or so and then you don't have to do any crazy iterator-collect allocations to call this with a block?

Reply:
> I would say it supports either filtered or unfiltered blocks rather being defined in terms of pre-filtered data.

Hmm, arguably, but why support anything other than a full block anyway? I'm not sure what te use-case is for filtering a pre-filtered block at all?

Reply:
That's a fair point. I'm re-thinking this after some of our offline discussions and will have a revised proposal for these interfaces.